poc-07.py

```diff
@@ -52,6 +55,9 @@ def isRead(self):
     def learnChanges(self, baseMessage):
         """Learn what was changed is which way."""
 
+        if self.isDelete() != baseMessage.idDelete():
```

Comment:
This is wrong. Deleted messages will be missing from storage. We'll never have message marked deleted here.